include/swift/AST/DiagnosticsSIL.def

```diff
@@ -43,6 +43,10 @@ ERROR(bridging_objcbridgeable_broken,none,
       "broken definition of '_ObjectiveCBridgeable' protocol: missing %0",
       (DeclName))
 
+ERROR(sil_function_redefinition,none,
+      "redefinition of function %0",
```

Comment:
I wonder if we could make the error message more specific, since it's not a redefinition of a function in the usual Swift sense, but an attempt to reuse the same symbol name. How about something like "multiple definitions of symbol name %0"?